docs/config.rst

```diff
@@ -0,0 +1,180 @@
+will also use such a token).
+
+
+What Needs to be Stored
+-----------------------
+
+As we are currently re-designing the storage mechanism, some of this
+might change. However, based on current understanding the state we
+WANT to store is:
+
+- for each magic-folder (indexed by "name"):
+  - local directory path (the "magic folder")
+  - an author:
+    - name
+    - private_key (32 bytes, a NaCl signing key)
+  - a write-cap of our mutable directory
+  - some state (see below)
+- the node-directory of our Tahoe-LAFS client
+  - we may need the `private/api_auth_token`
+  - we will need the `node.url` (at least)
+  - (it MAY be sufficient to only store the API endpoint (the contents
+    of `node.url`), not node-directory. If it's feasible to only store
+    the API endpoint, we should do that instead of caring about the
+    node-directory. It would probably still be convenient to refer to
+    a Tahoe client by node-directory (but we can just pull out the
+    Web-API endpoint). If we only store the endpoint and the Tahoe
+    config changes, the user would also have to change their
+    magic-folder config.
+- our own API endpoint (a Twisted server endpoint-string,
+  e.g. `tcp:1234` or `unix:/tmp/foo`)
+
+State that we need to store:
+
+- a secret token that protects the API (only users who can read the
+  file can access the endpoint)
+- list of other participants (per magic-folder) (see note)
+- local cache of remote snapshots (probably per-magic folder)
+- local copy of snapshots we haven't yet uploaded (probably per-magic folder)
+  - metadata details of the snapshot
+  - "stash directory" for file-data of the snapshot
+
+NOTE: Currently the "list of other participants" comes from the
+"collective dircap" but in the Leif Design each user decides whom they
+wish to pull updates from. So, the "list of other participants" above
+might be "a readcap of a directory" (and we would determine the actual
+list at runtime by reading that capability).
+
+
+UI to Change the Config
+-----------------------
+
+Writing configuration files is hard, error-prone and not a great
+interface. It also makes it hard (or impossible) to change where or
+how configuration is stored. All configuration shall be changed by an
+HTTP API. There will also be CLI commands that are thin wrappers of
+the HTTP API. GUI front-ends could use the command-line or the HTTP
+API. Nothing except the HTTP API should be touching the actual stored
+configuration.
+
+(Of course we can't stop users from editing the config files
+themselves, but this should be discouraged).
+
+
+.. _current:
+
+Where and How To Store the Configuration
+----------------------------------------
+
+Somewhat mirroring Tahoe-LAFS's design, configuration shall all be
+found in a single directory. This directory can be specified via
+command-line option. (Should we have a default? Seems sensible, most
+OSes have a reasonably well-defined place for programs to store
+configuration data).
```

Comment:
is there an API we can call, yet, that just returns the right path to use for all platforms? :/

Reply:
`#python` says https://pypi.org/project/appdirs/